internal/services/workloads/workloads_sap_virtual_instance_resource.go

```diff
@@ -0,0 +1,1129 @@
+									Type:     pluginsdk.TypeList,
+									Required: true,
+									ForceNew: true,
+									MaxItems: 1,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"instance_count": {
+												Type:     pluginsdk.TypeInt,
+												Required: true,
+												ForceNew: true,
+											},
+
+											"subnet_id": {
+												Type:         pluginsdk.TypeString,
+												Required:     true,
+												ForceNew:     true,
+												ValidateFunc: networkValidate.SubnetID,
+											},
+
+											"virtual_machine_configuration": SchemaForSAPVirtualInstanceVirtualMachineConfiguration(),
+
+											"database_type": {
+												Type:         pluginsdk.TypeString,
+												Optional:     true,
+												ForceNew:     true,
+												ValidateFunc: validation.StringInSlice(sapvirtualinstances.PossibleValuesForSAPDatabaseType(), false),
+											},
+
+											"disk_volume_configuration": SchemaForSAPVirtualInstanceDiskVolumeConfiguration(),
+										},
+									},
+								},
+
+								"full_resource_names": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									ForceNew: true,
+									MaxItems: 1,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"application_server": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"availability_set_name": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validate.AvailabilitySetName,
+														},
+
+														"virtual_machine": SchemaForSAPVirtualInstanceVirtualMachineFullResourceNames(),
+													},
+												},
+											},
+
+											"central_server": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"availability_set_name": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validate.AvailabilitySetName,
+														},
+
+														"load_balancer": SchemaForSAPVirtualInstanceLoadBalancerFullResourceNames(),
+
+														"virtual_machine": SchemaForSAPVirtualInstanceVirtualMachineFullResourceNames(),
+													},
+												},
+											},
+
+											"database_server": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"availability_set_name": {
```

Comment:
Can these be IDs instead 
```suggestion
														"availability_set_id": {
```

Reply:
It doesn't represent an existing tf resource so that I assume we can't do this.

Reply:
Is this because the API creates the availability set for the user? In which case [this comment](https://github.com/hashicorp/terraform-provider-azurerm/pull/22869#discussion_r1325369619) applies here as well, please verify that the API reliably cleans up any resources that it creates.

Reply:
Yes. Service API would create it for the user. Service API wouldn't delete it when SAP Virtual Instance is deleted. Because service team and major customers who are requesting this new resource confirmed that it's expected behavior and they expect to keep it since it's the part of the SAP system so that service API wouldn't actively delete it. But we can set "app_resource_group_name" and then SAP system will be deployed to this resource group. Then we could use "prevent_deletion_if_contains_resources = false" to delete the resources which created by service API when tf destroy.

Note: I already added note in [resource doc](https://github.com/hashicorp/terraform-provider-azurerm/pull/22869/files#diff-70507e2107861912a3d52f1986cbedbd0e7ea6c5ad2fc91af1af7ed049956d51R225) for this thing.

Reply:
I assume that we shouldn't design it to the required field "availability_set_id" since service team confirmed that we should not provide the name of the existing availability set for supporting below scenario 2 because service would create the availability set at the backend. As of the [availability set](https://github.com/Azure/azure-rest-api-specs/blob/927b6772aff66dc49b14f077d4b814d793a876ed/specification/workloads/resource-manager/Microsoft.Workloads/stable/2023-04-01/SAPVirtualInstance.json#L3166) property is used to specify or not specify the name of the availability set.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/19754191/3dd08d4f-5bc9-44b9-8853-5826a8b90f47)
 
Background:
SAP Virtual Instance is a resource that is used to manage/monitor the SAP System. So service team make this resource to support three kinds of scenarios.

Business scenarios:
1. Create SAP Virtual Instance based on the existing SAP System that is already provisioned and configured in advance in Azure. This kind of scenario is implemented via the "discovery_configuration" property in this PR. Note: The availability set, the application VM server, the disks and etc are parts of the SAP System.
2. Create SAP Virtual Instance based on the deployment definitions with OS configuration. This kind of scenario is implemented via the "deployment_with_os_configuration" property in this PR, which means service would automatically create SAP System that includes the availability set, the application VM server, the disks and etc.
3. Create SAP Virtual Instance based on the basic deployment definitions. Service team confirmed that this kind of scenario is replaced by above scenario 2 so that we don't need to support this scenario.